.github/workflows/gh-build.yml

```diff
@@ -54,57 +60,34 @@ jobs:
           role-duration-seconds: 28800 # 8 hours
           role-to-assume: arn:aws:iam::279114543810:role/gha-oidc-nv-legate
 
-      - name: Docker system prune
-        run: |
-          docker version
-          docker system prune --all --force
-
-      - name: Build docker image
+      - if: ${{ inputs.use-container }}
+        name: Build legate.core (in container)
         run: |
-          IMAGE_TAG=${{ env.IMAGE_NAME }}:${{ inputs.sha }}
-
-          continuous_integration/build-docker-image \
-            --base-image "$BASE_IMAGE" \
-            --image-tag "$IMAGE_TAG" \
-            --use-cuda "$USE_CUDA" \
-            --source-dir . \
-            --build-type "${{ inputs.build-type }}"
-
-      - name: Dump docker history of image before upload
-        run: |
-          IMAGE_TAG=${{ env.IMAGE_NAME }}:${{ inputs.sha }}
-          docker history $IMAGE_TAG
-
-      - if: env.PUSH_IMAGE == 'true'
-        name: Log in to container image registry
-        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
-
-      - if: env.PUSH_IMAGE == 'true'
-        name: Push image
-        run: |
-          IMAGE_TAG=${{ env.IMAGE_NAME }}:${{ inputs.sha }}
-
-          IMAGE_ID=ghcr.io/${{ github.repository_owner }}
-
-          # Change all uppercase to lowercase
-          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
-
-          IMAGE_ID=$IMAGE_ID/$IMAGE_TAG
-
-          docker tag $IMAGE_TAG $IMAGE_ID
-          docker push $IMAGE_ID
-
-      - name: Copy artifacts back to the host
+          set -xeuo pipefail
+
+          docker run \
+            -e AWS_REGION \
+            -e AWS_SESSION_TOKEN \
+            -e AWS_ACCESS_KEY_ID \
+            -e AWS_SECRET_ACCESS_KEY \
+            -e GITHUB_TOKEN \
+            -e USE_CUDA \
+            -v "$(pwd):$(pwd)" \
+            -v "$ARTIFACTS_DIR:$(pwd)/.artifacts" \
+            --rm "condaforge/miniforge3:latest" \
+            /bin/bash -c "cd $(pwd) && continuous_integration/scripts/build-legate ${{ inputs.build-type}}"
```

Comment:
Do we need to run docker directly? Isn't there a way to use the `container` feature of the github runner? It seems that we should run the same `build-legate` script, but either run it in a container or not, right?

Reply:
AFAIK the container feature cannot be conditionally switched off (say using if). So we will need to ...
1. Convert gh-build.yml in to a local action.
2. Create two different reusable workflows, one with container and one without and call the above local action from these workflows. 

The reusable part part come out to only 3 steps: checkout, build and upload. IMO the reusability is minimal (due to the limitations of the Github workflow markup language) and not worth the effort. Especially since the github container directive does pretty much  the same thing.

Reply:
I think it may make sense to do that in the long run. Basically, creating one workflow that does the work so to say, and then have different call sites for it. Right now, we have dispatch on `use-container` anyway, so I think we actually don't add anything if we do the dispatch in a different place. Yes, things would have to move around a bit, but the benefit would be a factored out build workflow that can be called from different contexts (e.g., different driver versions, different setups, etc.).